### PR TITLE
Refactor and reorganize javascript test traits

### PR DESCRIPTION
**Pull request by ol0lll**
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/thunder/thunder-distribution/blob/8.x-4.x/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.
- [ ] New features or changes are documented.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

File: tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php [hunk 1]
```diff
@@ -23,7 +24,8 @@ trait ThunderEntityBrowserTestTrait {
    *   Entity browser name.
    */
   public function openEntityBrowser(DocumentElement $page, $drupalSelector, $entityBrowser) {
```
Comment (ol0lll) [inline tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php]: Remove $page param.

File: tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php [hunk 2]
```diff
@@ -39,9 +41,22 @@ public function openEntityBrowser(DocumentElement $page, $drupalSelector, $entit
    *
    * @param \Behat\Mink\Element\DocumentElement $page
    *   Current active page.
+   * @param string $entityBrowser
+   *   Entity browser name.
    */
-  public function submitEntityBrowser(DocumentElement $page) {
-    $this->clickButtonDrupalSelector($page, 'edit-use-selected', FALSE);
+  public function submitEntityBrowser(DocumentElement $page, $entityBrowser) {
```
Comment (ol0lll) [inline tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php]: Remove $page param.

File: tests/src/FunctionalJavascript/ThunderArticleTestTrait.php [hunk 3]
```diff
@@ -33,9 +29,40 @@ public function articleFillNew(array $fieldValues) {
...
+  /**
+   * Expand all tabs on page.
+   *
+   * It goes up to level 3 by default.
+   *
+   * @param int $maxLevel
+   *   Max depth of nested collapsed tabs.
+   */
+  public function expandAllTabs($maxLevel = 3) {
+    $jsScript = 'jQuery(\'details.js-form-wrapper.form-wrapper:not([open]) > summary\').click().length';
```
Comment (ol0lll) [inline tests/src/FunctionalJavascript/ThunderArticleTestTrait.php]: Remove jquery

File: tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php [hunk 4]
```diff
@@ -37,11 +35,23 @@ public function openEntityBrowser(DocumentElement $page, $drupalSelector, $entit
...
+   * @param string $entityBrowser
+   *   Entity browser name.
    */
-  public function submitEntityBrowser(DocumentElement $page) {
-    $this->clickButtonDrupalSelector($page, 'edit-use-selected', FALSE);
+  public function submitEntityBrowser($entityBrowser) {
+    $page = $this->getSession()->getPage();
+    if ($entityBrowser == 'multiple_image_browser') {
+      $this->getSession()->wait(200);
+      $this->assertWaitOnAjaxRequest();
```
Comment (ol0lll) [inline tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php:45]: Do we need additional wait?

File: tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php [hunk 5]
```diff
@@ -37,11 +35,23 @@ public function openEntityBrowser(DocumentElement $page, $drupalSelector, $entit
...
+
+      $page->pressButton('Use selected');
+    }
+    elseif ($entityBrowser == 'image_browser') {
+      $page->pressButton('Select image');
+    }
+    elseif ($entityBrowser == 'video_browser') {
+      $page->pressButton('Select video');
+    }
```
Comment (ol0lll) [inline tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php:51]: Simplify conditions.

File: tests/src/FunctionalJavascript/ThunderMediaTestTrait.php [hunk 6]
```diff
@@ -20,45 +23,19 @@ trait ThunderMediaTestTrait {
...
-      $page->pressButton('Select video');
-      $element = 'iframe';
-    }
-
-    $this->getSession()->switchToIFrame();
-    $this->assertWaitOnAjaxRequest();
+    $this->submitEntityBrowser($entityBrowser);
 
-    $this->waitUntilVisible('div[data-drupal-selector="edit-' . str_replace('_', '-', $fieldName) . '-wrapper"] ' . $element);
+    $this->assertSession()->waitForElementVisible('css', 'div[data-drupal-selector="edit-' . str_replace('_', '-', $fieldName) . '-wrapper"]');
```
Comment (ol0lll) [inline tests/src/FunctionalJavascript/ThunderMediaTestTrait.php]: This is not the same, $element is missing

File: tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php (hunk 4 above)
Comment (chrfritsch) [reply tests/src/FunctionalJavascript/ThunderEntityBrowserTestTrait.php:45]: Yes

**Review by ol0lll — COMMENTED** [review]
Its fine.

**Review by chrfritsch — APPROVED** [review]

**Pull request merged**